configfile_generator.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+from dd4hep import Detector
+import re
+from collections import Counter
+from argparse import ArgumentParser
+from collections import defaultdict
+import pprint
+
+parser = ArgumentParser()
+
+parser.add_argument(
+    "--compactFile", help="DD4hep compact description xml", required=True
+)
+parser.add_argument(
+    "--maxDepth",
+    help="Maximum traversal depth of the detector tree",
+    default=10,
+    type=int,
+)
+parser.add_argument(
+    "--maxEdges", help="Maximum number of edges per connection", default=5, type=int
+)
+
+args = parser.parse_args()
+
+theDetector = Detector.getInstance()
+theDetector.fromXML(args.compactFile)
+
+# take part between last / and before the .xml
+detector_name = args.compactFile.split("/")[-1].split(".")[0]
+
+# start = theDetector.detector("OpenDataTracker")
+start = theDetector.world()
+
+def tree(detElement, depth):
```

Comment:
```suggestion
def tree(detElement, maxDepth=5, depth=0):
```
Passing in `maxDepth` as argument makes it possible to have this function being fully self contained and not depend on any other global / external values (like `args.maxDepth` at the moment). This makes it possible to import this function from another python module without any weird surprises because the value of that global variable might depend on whatever path is taken during imports.

Giving `depth` a default value of `0` makes it possible to simply call this function without it. Without this default value all users that call this will have to know that they need to pass in `0` here in order to get to their desired output.